processor/src/operations/decorators/mod.rs

```diff
@@ -7,9 +11,79 @@ impl Process {
     // DEBUGGING
     // --------------------------------------------------------------------------------------------
 
-    /// TODO: implement
-    pub fn op_debug(&self) -> Result<(), ExecutionError> {
-        unimplemented!();
+    /// Prints out debugging information based on options passed.
+    pub fn op_debug(&mut self, options: DebugOptions) -> Result<(), ExecutionError> {
+        match options {
+            DebugOptions::All => {
+                self.print_stack(None);
+                self.print_mem(None, None);
+            }
+            DebugOptions::Stack(n) => self.print_stack(n),
+            DebugOptions::Mem(n, m) => self.print_mem(n, m),
+            DebugOptions::Local(_) => self.print_local(),
+        }
+
```

Comment:
When we print out debug info, we should probably do a few additional things:

1. We should probably print out a clock cycle (from `system.clk()`) at which the debug operation is executed.
2. We should probably visually separate one debug op from another. Imagine a program which has several debug statements. Would be good to easily visually break them apart from each other.